### DISCUSSION TITLE
v13 SubscribeAndResolve obsolete - What is the future of observables?

### DISCUSSION THREAD

**Discussion by simonhoss** [ChilliCream/hotchocolate#5817] [Q&A]
Hey

We love HotChocolate a lot and thanks for all your work!

We're using Observables in combination with SubscribeAndResolve a lot in our backend. Now with v13 SubscribeAndResolve gots obsolete, therefore we asked ourself what is the future idea here? Is it still ok to bet on observables? Is there maybe another implementation we missed, so that we can still work with observables?

Thanks for your help!

**Comment by PascalSenn** [comment]
@simonhoss  Can you share an example of a resolver that uses SubscriveAndResolve? so we can show you how to do it now

**Reply by simonhoss** [reply]
Thanks for your answer.

A very simple example is this:

```csharp
public class Subscription
{
    [SubscribeAndResolve]
    public IObservable<int> GetCounter() => Observable.Interval(TimeSpan.FromSeconds(1)).Select(l => (int)l);
}
```

The point is, this examples returns an IObservable. We're using a lot IObservable as return for Subscriptions.

**Reply by simonhoss** [reply]
@PascalSenn any news here?

**Reply by simonhoss** [reply]
@PascalSenn or @michaelstaib sorry to ping you again. Can you help out here?
